code/modules/vehicles/multitile/multitile_bump.dm

```diff
@@ -221,6 +220,76 @@
 	qdel(src)
 	return TRUE
 
+/obj/structure/grille/handle_vehicle_bump(var/obj/vehicle/multitile/V)
+	if(!(V.vehicle_flags & VEHICLE_CLASS_MEDIUM || V.vehicle_flags & VEHICLE_CLASS_HEAVY))
+		V.move_momentum -= V.move_momentum * 0.5
+	visible_message(SPAN_DANGER("\The [V] crushes \the [src]!"))
+	playsound(src, 'sound/effects/grillehit.ogg', 20)
+	qdel(src)
+	return TRUE
+
+/obj/structure/inflatable/handle_vehicle_bump(var/obj/vehicle/multitile/V)
+	if(V.vehicle_flags & VEHICLE_CLASS_WEAK)
+		V.move_momentum -= V.move_momentum * 0.5
+	visible_message(SPAN_DANGER("\The [V] rams \the [src]!"))
+	density = FALSE
+	deflate(TRUE)
+	return TRUE
+
+/obj/structure/bed/chair/handle_vehicle_bump(var/obj/vehicle/multitile/V)
+	visible_message(SPAN_DANGER("\The [V] rams \the [src]!"))
+	if(stacked_size > 0)
+		stack_collapse()
+	else
+		qdel(src)
+	return TRUE
+
+/obj/structure/prop/dam/handle_vehicle_bump(var/obj/vehicle/multitile/V)
+	if(V.vehicle_flags & VEHICLE_CLASS_MEDIUM)
```

Comment:
not for weak?

Reply:
Only medium and heavy can ram those props, with only medium one getting slowdown.